Bottom: 4bf2ed177daee1cfe55c67c699398cadcc5474aa
Top:    51bb78519c545da2e4f27a2efe5e7764daff18e9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:32:28 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 040124c..545aaac 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1136,7 +1136,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
